source/extensions/omni.isaac.lab/omni/isaac/lab/sensors/frame_transformer/frame_transformer.py

```diff
@@ -136,9 +122,9 @@ def _initialize_impl(self):
             self._source_frame_offset_pos = source_frame_offset_pos.unsqueeze(0).repeat(self._num_envs, 1)
             self._source_frame_offset_quat = source_frame_offset_quat.unsqueeze(0).repeat(self._num_envs, 1)
 
-        # Keep track of mapping from the rigid body name to the desired frame, as there may be multiple frames
+        # Keep track of mapping from the rigid body name to the desired frames and prim path, as there may be multiple frames
         # based upon the same body name and we don't want to create unnecessary views
-        body_names_to_frames: dict[str, set[str]] = {}
+        body_names_to_frames: dict[str, dict[str, set[str] | str]] = {}
```

Comment:
NIT: Probably would require re-writing of the parsing logic but this seems to have become a very complex data structure. Would be good to make a note to simplify this in the future if it seems possible.

I suppose as long as it isn't causing run-time issues/performance drops, it's okay for now.

Reply:
The hope is that we can rely on IsaacSim team to re-implement this on IsaacSim side to simplify things quite a bit. I'm hesitant to refactor it before then, but agree it has gotten rather ugly with these new additions

Reply:
Did they say they will make something like this on their end? I am not aware of this discussion.